CIP Cardano Multichain Router b02eb0e3a2bf408a85bdbe6a4ce5603b.md

```diff
@@ -0,0 +1,395 @@
+
+(1) Key generation is used to create secret sharing ("keygen") without trusted dealers.
+
+(2) Use secret sharing, Paillier encryption and decryption to generate a signature ("signing").
+
+(3) Preprocessing data before generating signature.(“pre-sign”).
+
+When issuing the keygen/signing request command, there are two modes to choose from:
+
+(1) Each participant node needs to approve the request command with its own account.It will first get the request command from the local approval list, and then approve or disagree.
+
+(2) Each participant node does not need to approve the request command, which is agreed by default.
+
+The SMPC network is responsible for signing transactions on each supported blockchain to perform a multitude of tasks comprising the management of asset accounts and smart contracts. A Threshold, will be needed to sign the transaction. This Threshold Signature (TSS) is denoted as (t, n), such as (9,15), (15,21), (21,31) etc., where t is the number of nodes needed to sign, and n is the total number of nodes in the set.
+
+## Router
+
+Cardano will be integrated with Multichain router V3, which was lauched on June 4th 2021. The Router can allow assets to be transferred among all others connected chains including Cardano. Check more about router here:
+
+[https://docs.multichain.org/getting-started/how-it-works/cross-chain-router](https://docs.multichain.org/getting-started/how-it-works/cross-chain-router)
+
+# User Cases
+
+Tokens on different chains can be bridged to Cardano, but let’s simplify it to the scenario between Ethereum and Cardano.
+
+On Cardano side, the Multichan SMPC nodes manage a MPC address, which locks the Cardano native tokens bridged to Ethereum and mint the wrapped tokens bridged from Ethereum.
+
+On Ethereum side, the Multichain SMPC nodes will call Multichain router smart contract to send the Ethereum native token in the pool to users and mint the wrapped tokens bridged from Cardano. 
+
+## A. ****Bridging Ethereum native tokens from Ethereum to Cardano****
+
+1, User send USDC to the USDC pool on Ethereum and indicate the receiving address on Cardano.
+
+2, The SMPC nodes verify the user transaction on Ethereum independently.
+
+3, After the verification, the SMPC nodes work together to sign a minting transaction on Cardano.
+
+4, The SMPC nodes mint wrapped USDC to user’s address on Cardano. 
+
+## B. Bridging wrapped Ethereum native token from Cardano to Ethereum
+
+1, User send wrapped USDC to the SMPC address on Cardano and indicate the receiving address on Ethereum.
+
+2, The SMPC nodes verify the user transacton on Cardano independently.
+
+3, After the verification, the SMPC nodes work together to sign a transfer transaction on Ethereum.
+
+4, The SMPC nodes send the native USDC to user’s address on Ethereum. 
+
+## C. Bridging Cardano native token from Cardano to Ethereum
+
+1, User send native ADA to the SMPC address on Cardano and indicate the receiving address on Ethereum.
```

Comment:
Since the user can just send any tokens/ada here. It'd be good to define how those tokens will be handled.